### PR TITLE
LPS-123845 Remove the frontend OData parser and use a backend parser instead

### PR DESCRIPTION
**Pull request by dgarciasarai**
**Description**
We want to add IN and NOT IN operators in the Segments Editor but the library that we were using in the frontend side called `odata-v4-parser` does not support those operations. Also, the library is not up-to-date, the last commits are from 2018.

**Solution**
Remove the parser from frontend `odata-v4-parser` and use a backend one.

**How to test it**
Go to People > Segments and everything should work as before but now the Segments Editor uses a backend parser.

### CODE REVIEW

**Comment by dgarciasarai** [comment]
Hey @liferay-frontend, can you give me feedback about the frontend part? Backend code is going to be reviewed in @liferay-tango :) After both reviews, we will forward the changes from @liferay-tango. Thanks! cc: @cgoncas

**Comment by dgarciasarai** [comment]
I rebased the pull request and fix an LPS in one commit.

**Review by julien — APPROVED** [review]

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:stop

**Comment by dgarciasarai** [comment]
Thanks @julien for the review :) Let's wait for another reviewer today. Tomorrow I'll close the pull request and prepare the final PR for Brian

File: modules/apps/segments/segments-web/test/js/utils/odata.es.js [hunk 1]
```diff
@@ -58,187 +72,333 @@ describe('odata-util', () => {
...
 			);
 		});
-	});
 
-	describe('translateQueryToCriteria', () => {
-		it('translates a query string into a criteria map', () => {
-			expect(
-				ODataUtil.translateQueryToCriteria("(firstName eq 'test')")
-			).toEqual({
+		it('is able to translate a query string to map and back to string', () => {
```
Comment (wincent) [inline modules/apps/segments/segments-web/test/js/utils/odata.es.js:76]: (non-blocking feedback)

You can make some of these shorter and therefore easier to fit on one line by dropping the "is able to". eg.

```js
it('translates a query string to map and back', () => {
```

**Review by wincent — COMMENTED** [review]
JS changes all look fine to me (mostly deletions). Didn't look at the scary Java that I am not qualified to look at.

File: modules/apps/segments/segments-web/test/js/utils/odata.es.js (hunk 1 above)
Comment (dgarciasarai) [reply modules/apps/segments/segments-web/test/js/utils/odata.es.js:76]: Great @wincent, thank you so much for the feedback. I'll change it.

**Comment by dgarciasarai** [comment]
This is the easiest one...There is more frontend code in the way @wincent ;) I will apply your feedback about the tests.

**Comment by cgoncas** [comment]
Hey @dgarciasarai !!!

I have updated this pull with the changes that Brian already merged on master. 

Also, I have added the two missing commits :).

Thanks!!!

**Comment by dgarciasarai** [comment]
Thanks @julien @wincent and @cgoncas!! I'm going to apply the feedback about the tests and send the changes to @liferay-tango for a final review. Thanks all for your feedback and help :)

**Comment by dgarciasarai** [comment]
Continues in https://github.com/liferay-tango/liferay-portal/pull/598.

**Pull request closed without merging**